sites/target.py

```diff
@@ -0,0 +1,199 @@
+
+        result = re.search(b"[$]cdc_[a-zA-Z0-9]{22}_", data)
+
+        if result is not None:
+            self.status_signal.emit(self.create_msg("Changing value in Chromedriver", "normal"))
+            data = data.replace(result.group(0), val.encode())
+            fin.close()
+            fin = open(loc, 'wb')
+            fin.truncate()
+            fin.write(data)
+            fin.close()
+        else:
+            fin.close()
+
+    def init_driver(self):
+        driver_manager = ChromeDriverManager()
+        driver_manager.install()
+        self.change_driver(driver_path)
+        browser = webdriver.Chrome(driver_path)
+
+        browser.execute_cdp_cmd("Page.addScriptToEvaluateOnNewDocument", {
+            "source": """
+                  Object.defineProperty(navigator, 'webdriver', {
+                   get: () => undefined
+                  })
+                """
+        })
+
+        return browser
+
+    def login(self):
+        self.browser.get("https://www.target.com")
+
+        self.browser.find_element_by_id("account").click()
+
+        wait(self.browser, 3).until(EC.element_to_be_clickable((By.ID, "accountNav-signIn"))).click()
+
+        my_username = "username"
+        my_password = "password"
+
+        wait(self.browser, 3).until(EC.presence_of_element_located((By.ID, "username"))).send_keys(my_username)
+        password = self.browser.find_element_by_id("password")
+        password.send_keys(my_password)
+        self.browser.find_element_by_id("login").click()
+
+        # Gives it time for the login to complete
+        time.sleep(random_delay(self.monitor_delay, settings.random_delay_start, settings.random_delay_stop))
+
+    def create_msg(self, msg, status):
+        return {"msg": msg, "status": status}
+
+    def monitor(self):
+        img_found = False
+        in_stock = False
+
+        self.browser.get(self.product)
+
+        while not in_stock:
+            try:
+                if not img_found:
+                    product_img = self.browser.find_elements_by_class_name('slideDeckPicture')[0].find_element_by_tag_name(
+                        "img")
+                    self.image_signal.emit(product_img.get_attribute("src"))
+                    self.product_image = product_img.get_attribute("src")
+                    img_found = True
+            except Exception as e:
+                continue
+
+            try:
+                ship_btn = self.browser.find_element_by_xpath('//button[@data-test= "shipItButton"]')
+                self.browser.execute_script("return arguments[0].scrollIntoView(true);", ship_btn)
+                ship_btn.click()
+                in_stock = True
+                self.status_signal.emit(self.create_msg("Added to cart", "normal"))
+            except Exception as e:
+                self.status_signal.emit(self.create_msg("Waiting on Restock", "normal"))
+                self.browser.refresh()
```

Comment:
You should add a time.sleep here so that the bot has a delay between refreshes

```suggestion
                self.status_signal.emit(self.create_msg("Waiting on Restock", "normal"))
                time.sleep(random_delay(self.monitor_delay, settings.random_delay_start, settings.random_delay_stop))
                self.browser.refresh()
```